crates/services/consensus_module/poa/src/verifier.rs

```diff
@@ -0,0 +1,78 @@
+use anyhow::ensure;
+use fuel_core_storage::Result as StorageResult;
+use fuel_core_types::{
+    blockchain::{
+        block::Block,
+        header::BlockHeader,
+        primitives::BlockHeight,
+    },
+    fuel_types::Bytes32,
+};
+
+#[cfg(test)]
+mod tests;
+
+/// The config of the block verifier.
+pub struct Config {
+    /// If the manual block is enabled, skip verification of some fields.
+    pub enabled_manual_blocks: bool,
+}
+
+#[cfg_attr(test, mockall::automock)]
+/// The port for the database.
+pub trait Database {
+    /// Gets the block header at `height`.
+    fn block_header(&self, height: &BlockHeight) -> StorageResult<BlockHeader>;
+
+    /// Gets the block header BMT MMR root at `height`.
+    fn block_header_merkle_root(&self, height: &BlockHeight) -> StorageResult<Bytes32>;
+}
+
+pub fn verify_poa_block_fields<D: Database>(
+    config: &Config,
+    database: &D,
+    block: &Block,
+) -> anyhow::Result<()> {
+    let height = *block.header().height();
+    ensure!(
+        height != 0u32.into(),
+        "The PoA block can't have the zero height"
+    );
+
+    let prev_height = height - 1u32.into();
+    let prev_root = database.block_header_merkle_root(&prev_height)?;
+    let header = block.header();
+    ensure!(
+        header.prev_root() == &prev_root,
+        "The genesis time should be unix epoch time"
```

Comment:
Unsure how this check relates to epoch time 🤔

Reply:
seems like a copy-paste bug from genesis verification, applied a suggestion to fix it.